git/revparse.go

```diff
@@ -193,19 +257,27 @@ func RevParse(c *Client, opt RevParseOptions, args []string) (commits []ParsedRe
 			} else {
 				var sha string
 				var exclude bool
-				var err error
 				if arg[0] == '^' {
 					sha = arg[1:]
 					exclude = true
 				} else {
 					sha = arg
 					exclude = false
 				}
-				cmt, err := RevParseCommit(c, &opt, sha)
-				if err != nil {
-					err2 = err
+				if path := strings.Index(arg, ":"); path >= 0 {
```

Comment:
Maybe strings.Contains() would be a bit clearer?

Reply:
Sure, changed